### DISCUSSION TITLE
Suggestion on graph.hpp

### DISCUSSION THREAD

**Discussion by mrprajesh** [ashwinktpu/StarPlat#9] [Ideas]
I had a few suggestions. Just thought of logging here. Maybe we can use if it useful.
1. We can use a `short`, `int8_t` or non `32 bit int` for storing edge `weight`. 
https://github.com/nibeditabh/GraphDSL/blob/fe47bab3a9c35526264abd4cf2b418b98e84d22b/graphcode/graph.hpp#L18

As all our graphs are unweighted, this will bring down memory allocation by at least half for large value of `N` and possibly at many other places where `weight` is used. 
https://github.com/nibeditabh/GraphDSL/blob/fe47bab3a9c35526264abd4cf2b418b98e84d22b/graphcode/graph.hpp#L421

2. We could have a command line argument for knowing if the input we are supplying is a `directed` or `undirected`. Maybe it will help us in generation of code appropriately. 
```
./finalcode -f <dsl filename> -b <backend> -d    # -d means directed. by default it is undirected.
```
Eg. In BC if a `div by 2` is required or not. Maybe we can discuss. @rupesh0508 
https://github.com/nibeditabh/GraphDSL/blob/8055d70fde1888159130afc0ea8f95e9bfe4b1af/graphcode/cudaManual/bcManualv6-sync.cu#L87

**Comment by rupesh0508** [comment]
1. Seems reasonable if all weights are small. @mrprajesh, go ahead and update if you find max-weight small in our 10 graphs. Using a typedef would help reduce the number of changes.
2. @nibeditabh, what do you suggest?

**Reply by nibeditabh** [reply]
Yeah, this seems reasonable. We can work on this.
